### PR TITLE
VACMS-4049: Policy-Tooltips-Accessible

### PR DESCRIPTION
**Pull request by kaise-lafrai**
## Description

See _issueid_. https://app.zenhub.com/workspaces/vagov-cms-team-5c0e7b864b5806bc2bfc2087/issues/department-of-veterans-affairs/va.gov-cms/4049

## QA steps

- Navigate to preview environment
- Navigate here: `/admin/structure/types/manage/vamc_system_policies_page/form-display`
---
- Confirm the following **3** new `Tooltip` field groups:
- [ ] National top of page content
- [ ] National general visitation policy
- [ ] National bottom of page content
---
- Select the gear icon for each of the field groups from above and confirm the following fields:
- [ ]
image
---
- On a separate tab navigate here: `/node/add/vamc_system_policies_page`
- [ ] Confirm the tooltip appears and that the text within the tooltip matches the `Tooltip Description` from the image above.
- [ ] Complete form, save and confirm the Tooltips appear and function as expected.
- [ ]  using the tab button on your keyboard validate that the tooltip activates with tab and reflects the order of the content.
- [ ] Inspect element to confirm accessibility issues are resolved.
---
- Navigate back to the manage form display tab
- Select the gear setting next for `National top of page content`
- Change the `Show Label` field to `Yes`
- Copy the `Tooltip Description` over to the `Description` textarea. 
- Save
---
- Navigate back here: `/node/add/vamc_system_policies_page`
- [ ] Confirm the tooltip does NOT appear.
- [ ] Confirm the description now appears under the field label.
- [ ] Inspect element to confirm accessibility issues are resolved.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      '#access' => FALSE,
+    ];
+
+    $form['effect'] = [
+      '#access' => FALSE,
+    ];
+
+    $form['speed'] = [
+      '#access' => FALSE,
+    ];
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:163]: curious:  what are these for?  I'm not familiar with these.

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultContextSettings($context) {
+    $defaults = [
+      'open' => FALSE,
+      'required_fields' => $context == 'form',
+    ] + parent::defaultSettings($context);
+
+    if ($context == 'form') {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:199]: Suggestion:   Whenever you aren't sure of the values that may come through the variable, you want to be defensive by using the strict comparison (===) It is a subtle difference, but always safer to default to strict or your risk unintended results.  (see the comparison of the truth tables.
image

image

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Overall looks good Kaise.  Just a couple of minor changes.

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php (hunk 1 above)
Comment (kaise-lafrai) [reply docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:163]: These hide some of the unnecessary fields that come with Html element.

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php (hunk 2 above)
Comment (kaise-lafrai) [reply docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:199]: Fixed in new commit.

File: config/sync/core.entity_form_display.node.vamc_system_policies_page.default.yml [hunk 3]
```diff
@@ -61,40 +61,55 @@ third_party_settings:
         - field_cc_top_of_page_content
       parent_name: ''
       weight: 0
-      format_type: fieldset
+      format_type: Tooltip
```
[the remark this answers is not in the document]
Comment (kaise-lafrai) [reply config/sync/core.entity_form_display.node.vamc_system_policies_page.default.yml:64]: Resolved in new commit

File: docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\field_group\Plugin\field_group\FieldGroupFormatter\HtmlElement;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Component\Utility\Html;
+
+/**
+ * Tooltip field group formatter.
+ *
+ * @FieldGroupFormatter(
+ *   id = "Tooltip",
```
[the remark this answers is not in the document]
Comment (kaise-lafrai) [reply docroot/modules/custom/va_gov_backend/src/Plugin/field_group/FieldGroupFormatter/Tooltip.php:15]: resolved in new commit

File: config/sync/core.entity_form_display.node.vamc_system_policies_page.default.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaise-lafrai) [reply config/sync/core.entity_form_display.node.vamc_system_policies_page.default.yml:64]: Take that back: I couldn't figure out why my format type starts with a capital T. I tried changing it and it breaks the fieldgroup selection on the content type configration.

**Comment by kaise-lafrai** [comment]
@swirtSJW @kevwalsh Ready for review.

**Comment by swirtSJW** [comment]
Looks like the partial switch of the tooltip manchine name to lower case and the config in uppercase led to a mismatch on import.
image

I pushed up a fix to sync the two.

My guess is it did not work locally for you due to caching.  This needed a cache clear before config import.

**Review by swirtSJW — APPROVED** [review]
This looks great Kaise.  Outstanding work.   I added the role attribute.

Just waiting on tests to pass to merge this.

**Pull request merged**